qiskit/pulse/utils.py

```diff
@@ -16,9 +16,62 @@
 Pulse utilities.
 """
 import warnings
+
+from collections import namedtuple
+from typing import List
+
+from .exceptions import PulseError
+
 # pylint: disable=unused-argument
 
 
+Interval = namedtuple('Interval', ['start', 'stop'])
+"""Intervals [start, stop) are exclusive of the stop time."""
+
+
+def overlaps(first: Interval, second: Interval) -> bool:
+    """
+    Return True iff first and second overlap.
+
+    Note: first.stop may equal second.start, since Interval stop times are exclusive.
+    """
+    if first.start == second.start == second.stop:
+        # They fail to overlap if one of the intervals has duration 0
```

Comment:
Ah, `first.start == first.end` case is covered in Line 43.. I think there is no bug here, but how about adding tests including zero duration intervals like below for safety in the future?
```
# For some integers a, b (a < b)
assertFalse(has_overlap(Interval(a,a), Interval(a,a)))
assertFalse(has_overlap(Interval(a,a), Interval(a,b)))
assertFalse(has_overlap(Interval(a,b), Interval(a,a)))
assertFalse(has_overlap(Interval(a,b), Interval(b,b)))
assertFalse(has_overlap(Interval(b,b), Interval(a,b)))
assertTrue(has_overlap(Interval(a,a+2), Interval(a+1,a+1)))
assertTrue(has_overlap(Interval(a+1,a+1), Interval(a,a+2)))
```